### DISCUSSION TITLE
Docker arm64 support

### DISCUSSION THREAD

**Discussion by cracyfloyd** [johanmeijer/grott#276] [Q&A] [answered]
Iam not shure iam right here to ask for docker requests.
but i try.
it is possible to make a a docker for arm 64bit. 
i use debian bullseye 64bit as main system in a vm. in this vm are running all my smarthome stuff like iobroker, homebridge etc...
i am still missing that i can implement this nice grott feature in my smarthome system (VM) directly. for that i can use grott now i use docker desktop for mac (arm). there run the rpi docker 32 bit.
thnx

**Comment by johanmeijer** [comment, marked as answer]
At this moment I do not have a arm64 server running to build and test a docker container. In the past I had my  rpi 4 running a 64 bit os and has been able to run the grottrpi Version (arm32) on that (creating a docker container on arm64 and run on arm32 was not possible).

Can you try the grottrpi (or grottserverrpi) image ? Otherwise I can publish the latest dockerfile and you can create your own image. 

I do understand that it should be possible to create docker images directly from github. I have planned to look into that soon.

**Reply by cracyfloyd** [reply]
I try the 32bit rpi image under debian 64bit, but it doesnt start. I do not have try to build an own image until now... Huuhhh...

but i can try.

thnx a lot.

**Reply by egguy** [reply]
If you are interested, I did some tests a long time ago to build multi arch image. They are available on https://hub.docker.com/repository/docker/egguy/grott-ci-test and https://hub.docker.com/repository/docker/egguy/grottserver-ci-test.

They are available in armv7 arm64 and amd64 edition.

The code is available on the branch: https://github.com/egguy/grott/tree/feature/docker-build if you are interested.

There are some problems in the arm edition and grottserver image. The server requires libscrc to be compiled.
On python 3.7-slim image it's not available, when using on the normal image it works, but it's 250 Mo vs 1200Mo.

Would it be possible to use a python version fallback? So, it could build without the dependency?

**Comment by cracyfloyd** [comment]
Good morning.

I have running for 2 Weeks now your 64bit version without complications (without server). Thnx Thnx Thnx a lot.

**Comment by mameyer1** [comment]
Hi,
I'm facing the same "problem"  with the MIN XH inverter. If i can help, let me know

Marcel
